packages/twenty-server/src/engine/core-modules/onboarding/onboarding.service.ts

```diff
@@ -127,35 +63,82 @@ export class OnboardingService {
       return OnboardingStatus.WORKSPACE_ACTIVATION;
     }
 
-    if (await this.isProfileCreationOnboardingStatus(user)) {
+    const userVars = await this.userVarsService.getAll({
```

Comment:
@charlesBochet haven't tested but here is my suggestion for more readability and scalability
```typescript
  const userVars = await this.userVarsService.getAll({
    userId: user.id,
    workspaceId: user.defaultWorkspaceId,
  });

  const stepsMap = new Map<OnboardingStepKeys, OnboardingStatus>([
    [OnboardingStepKeys.ONBOARDING_CREATE_PROFILE_COMPLETE, OnboardingStatus.PROFILE_CREATION],
    [OnboardingStepKeys.ONBOARDING_CONNECT_ACCOUNT_COMPLETE, OnboardingStatus.SYNC_EMAIL],
    [OnboardingStepKeys.ONBOARDING_INVITE_TEAM_COMPLETE, OnboardingStatus.INVITE_TEAM],
  ]);

  for (const [step, status] of stepsMap) {
    if (!userVars.get(step)) {
      return status;
    }
  }

  return OnboardingStatus.COMPLETED;
```
Map "should" keep insertion order from what I'm reading but if you don't see any benefit from this syntax don't bother!

Reply:
I think this will be too rigid and we might have a more complex logic in the future where several userVars decide the onboardingStep